Publishing to GitHub Package Registry failing with 401 Unauthorized

## 🚨 Publishing Failure - GitHub Package Registry

### Error Details
The package publishing to GitHub Package Registry is failing with a 401 Unauthorized error:

```
npm notice Publishing to https://npm.pkg.github.com with tag latest and default access
npm error code E401
npm error 401 Unauthorized - PUT https://npm.pkg.github.com/@signal-meaning%2fdeepgram-voice-interaction-react - unauthenticated: User cannot be authenticated with the token provided.
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2025-10-21T16_25_28_081Z-debug-0.log
Error: Process completed with exit code 1.
```

### Issue Type
- **Type**: Regression (was working previously)
- **Severity**: High (blocks v0.4.0 release)
- **Component**: GitHub Actions / Package Publishing

### Root Cause Analysis
This appears to be an authentication issue with the GitHub Package Registry. Possible causes:

1. **Token Expiration**: The `GITHUB_TOKEN` may have expired or been revoked
2. **Token Permissions**: The token may not have the required `packages:write` permission
3. **Registry Configuration**: The npm registry configuration may be incorrect
4. **Package Scope**: There may be an issue with the `@signal-meaning` scope configuration

### Investigation Needed
- [ ] Check GitHub Actions secrets and token permissions
- [ ] Verify `publishConfig` in `package.json`
- [ ] Review GitHub Package Registry settings
- [ ] Check if the package name/scope is correctly configured
- [ ] Verify the workflow has the correct `permissions` section

### Workflow File
The issue is occurring in `.github/workflows/publish.yml`:

```yaml
- name: Publish to GitHub Package Registry
  run: npm publish
  env:
    NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### Expected Behavior
The package should publish successfully to `https://npm.pkg.github.com/@signal-meaning/deepgram-voice-interaction-react`

### Actual Behavior
Getting 401 Unauthorized error, preventing package publication.

### Impact
- **Blocks**: v0.4.0 release
- **Affects**: Package distribution and availability
- **Users**: Cannot install the latest version

### Related
- This was working previously, indicating a regression
- Test fixes were just merged (PR #131) but publishing is now failing
- Need to resolve before v0.4.0 release can proceed

### Next Steps
1. Investigate token permissions and configuration
2. Check GitHub Package Registry settings
3. Verify workflow permissions
4. Test publishing locally if possible
5. Update token or configuration as needed

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -27,6 +27,7 @@ jobs:
           node-version: '20'
           registry-url: 'https://npm.pkg.github.com'
           scope: '@signal-meaning'
+          token: ${{ secrets.GITHUB_TOKEN }}
           
       - name: Install dependencies
         run: npm ci
@@ -39,8 +40,6 @@ jobs:
         
       - name: Publish to GitHub Package Registry
         run: npm publish
-        env:
-          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           
       - name: Verify package installation
         run: |

--- a/package.json
+++ b/package.json
@@ -51,7 +51,8 @@
   "author": "Deepgram",
   "license": "MIT",
   "publishConfig": {
-    "registry": "https://npm.pkg.github.com"
+    "registry": "https://npm.pkg.github.com",
+    "@signal-meaning:registry": "https://npm.pkg.github.com"
   },
   "devDependencies": {
     "@babel/core": "^7.28.4",

